### PR TITLE
DAOS-15966 control: Remove dmg storage query device-health (#14508)

### PR DESCRIPTION
**Pull request by tanabarr**
Remove unnecessary command alias in favor of dmg storage query list-devices --health.

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by daltonbohning — APPROVED** [review]

File: src/include/daos/tests_lib.h [hunk 1]
```diff
@@ -514,7 +514,7 @@ int dmg_storage_set_nvme_fault(const char *dmg_config_file,
  * Get NVMe Device health stats.
  *
  * \param[in] dmg_config_file	DMG config file
- * \param[in] host		Get device-health from the given host.
+ * \param[in] host		Get list-devices --health from the given host.
```
Comment (kjacque) [inline src/include/daos/tests_lib.h:517]: Just noticed this mistaken change, looks like a find/replace change. I guess this must have been in the master PR too.

**Review by kjacque — APPROVED** [review]
Just a minor issue in a doxygen comment, not blocking on it but it would be ideal to fix both here and in master.

File: src/include/daos/tests_lib.h (hunk 1 above)
Comment (tanabarr) [reply src/include/daos/tests_lib.h]: fixed this and will fix master

**Review by daltonbohning — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Comment by tanabarr** [comment]
OSAOnlineReintegration CI failure is unrelated to change

**Comment by daltonbohning** [comment]
FYI 2.6 is closed

**Comment by daltonbohning** [comment]
> OSAOnlineReintegration CI failure is unrelated to change

Looks like https://daosio.atlassian.net/browse/DAOS-15608

**Pull request merged**